### PR TITLE
Profile controller

### PR DESCRIPTION
**Pull request by Greengrer**

### CODE REVIEW

File: src/main/java/com/vinylteam/vinyl/security/SecurityConfiguration.java [hunk 1]
```diff
@@ -16,12 +16,10 @@
...
 import org.springframework.security.crypto.password.Pbkdf2PasswordEncoder;
 import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
-import org.springframework.web.cors.CorsConfiguration;
-import org.springframework.web.cors.CorsConfigurationSource;
-import org.springframework.web.cors.UrlBasedCorsConfigurationSource;
 import org.springframework.web.servlet.config.annotation.CorsRegistry;
 import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
-import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
+
+
```
Comment (Greengrer) [inline src/main/java/com/vinylteam/vinyl/security/SecurityConfiguration.java]: extra lines should be removed

**Review by Greengrer — COMMENTED** [review]
Just a couple exatra lines in one of the files so far

File: src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java [hunk 2]
```diff
@@ -100,4 +108,21 @@ public void setUserStatusTrue(long id) {
         namedParameterJdbcTemplate.update(UPDATE_USER_STATUS, sqlParameterSource);
     }
 
+    @Override
+    public void changeProfile(User user, String email, String discogsUserName) {
+        MapSqlParameterSource sqlParameterSource = new MapSqlParameterSource()
+                .addValue("new_email", email)
+                .addValue("discogs_user_name", discogsUserName)
+                .addValue("email", user.getEmail());
+        namedParameterJdbcTemplate.update(UPDATE_PROFILE_FIELDS, sqlParameterSource);
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java]: please remove variable declaration

File: src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java [hunk 3]
```diff
@@ -100,4 +108,21 @@ public void setUserStatusTrue(long id) {
...
+    }
+
+    @Override
+    public void changeUserPassword(User user) {
+        MapSqlParameterSource sqlParameterSource = new MapSqlParameterSource()
+                .addValue("email", user.getEmail())
+                .addValue("password", user.getPassword());
+        namedParameterJdbcTemplate.update(UPDATE_PASSWORD, sqlParameterSource);
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java]: please remove variable declaration

File: src/main/java/com/vinylteam/vinyl/exception/JwtAuthenticationException.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.vinylteam.vinyl.exception;
+
+import org.springframework.security.core.AuthenticationException;
+
+public class JwtAuthenticationException extends AuthenticationException {
+    public JwtAuthenticationException(String msg, Throwable t) {
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/exception/JwtAuthenticationException.java]: bad parameters names

File: src/main/java/com/vinylteam/vinyl/exception/JwtAuthenticationException.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+        super(msg, t);
+    }
+
+    public JwtAuthenticationException(String msg) {
+        super(msg);
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/exception/JwtAuthenticationException.java]: bad parameter name

File: src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserService.java [hunk 6]
```diff
@@ -120,6 +118,42 @@ public User findByEmail(String email) {
...
+    @Override
+    public User changeUserPassword(UserInfoRequest changeRequest, User user) {
+        String oldPassword = changeRequest.getPassword();
+        if (!isNotEmptyNotNull(oldPassword)) {
+            throw new RuntimeException(UserErrors.EMPTY_PASSWORD_ERROR.getMessage());
+        }
+        boolean checkOldPassword = securityService.validateIfPasswordMatches(user, oldPassword.toCharArray());
+        if (!checkOldPassword) {
+            throw new BadCredentialsException(UserErrors.WRONG_PASSWORD_ERROR.getMessage());
+            //throw new RuntimeException(UserErrors.WRONG_PASSWORD_ERROR.getMessage());
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserService.java]: YAGNI

File: src/main/java/com/vinylteam/vinyl/service/impl/JwtTokenProvider.java [hunk 7]
```diff
@@ -81,7 +84,8 @@ public boolean isTokenValid(String token) {
             return !tokenStorageService.isTokenBlocked(token);
         } catch (JwtException | IllegalArgumentException e) {
             log.error("JWT token is expired or invalid", e);
-            return false;
+            //return false;
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/service/impl/JwtTokenProvider.java]: YAGNI

**Review by hemiSRT8 — CHANGES_REQUESTED** [review]

**Review by hemiSRT8 — APPROVED** [review]

**Pull request merged**